cluster-autoscaler/core/static_autoscaler_test.go

```diff
@@ -1377,6 +1380,103 @@ func TestSubtractNodes(t *testing.T) {
 	}
 }
 
+func TestFilterOutYoungPods(t *testing.T) {
+	now := time.Now()
+	klog.InitFlags(nil)
+	flag.CommandLine.Parse([]string{"--logtostderr=false"})
+
+	p1 := BuildTestPod("p1", 500, 1000)
+	p1.CreationTimestamp = metav1.NewTime(now.Add(-1 * time.Minute))
+	p2 := BuildTestPod("p2", 500, 1000)
+	p2.CreationTimestamp = metav1.NewTime(now.Add(-1 * time.Minute))
+	p2.Annotations = map[string]string{
+		podScaleUpDelayAnnotationKey: "5m",
+	}
+	p3 := BuildTestPod("p2", 500, 1000)
+	p3.CreationTimestamp = metav1.NewTime(now.Add(-1 * time.Minute))
+	p3.Annotations = map[string]string{
+		podScaleUpDelayAnnotationKey: "2m",
+	}
+	p4 := BuildTestPod("p2", 500, 1000)
+	p4.CreationTimestamp = metav1.NewTime(now.Add(-1 * time.Minute))
+	p4.Annotations = map[string]string{
+		podScaleUpDelayAnnotationKey: "error",
+	}
+
+	tests := []struct {
+		name               string
+		newPodScaleUpDelay time.Duration
+		runTime            time.Time
+		pods               []*apiv1.Pod
+		expectedPods       []*apiv1.Pod
+		expectedError      string
+	}{
+		{
+			name:               "annotation delayed pod now",
+			newPodScaleUpDelay: 0,
+			runTime:            now,
+			pods:               []*apiv1.Pod{p1, p2},
+			expectedPods:       []*apiv1.Pod{p1},
+		},
+		{
+			name:               "annotation delayed pod after delay",
+			newPodScaleUpDelay: 0,
+			runTime:            now.Add(5 * time.Minute),
+			pods:               []*apiv1.Pod{p1, p2},
+			expectedPods:       []*apiv1.Pod{p1, p2},
+		},
+		{
+			name:               "globally delayed pods",
+			newPodScaleUpDelay: 5 * time.Minute,
+			runTime:            now,
+			pods:               []*apiv1.Pod{p1, p2},
+			expectedPods:       []*apiv1.Pod{},
+		},
+		{
+			name:               "annotation delay smaller than global",
+			newPodScaleUpDelay: 5 * time.Minute,
+			runTime:            now,
+			pods:               []*apiv1.Pod{p1, p3},
+			expectedPods:       []*apiv1.Pod{p1},
+			expectedError:      "Failed to set pod scale up delay for",
+		},
+		{
+			name:               "annotation delay with error",
+			newPodScaleUpDelay: 0,
+			runTime:            now,
+			pods:               []*apiv1.Pod{p1, p4},
+			expectedPods:       []*apiv1.Pod{p1},
```

Comment:
I don't think pods with malformed annotation should be ignored. I think logging an error should be enough - if annotation cannot be parsed, CA behavior should fall back to global setting instead of banning the pod forever.

Reply:
There was a bug in my tests, so it didn't catch that my code reformatting commit actually broke this behavior. Now I reverted code reformatting (pretty much completely) and fixed the tests.